### PR TITLE
refactor(ui): Extract create run form's logic to helper functions

### PR DESCRIPTION
**Pull request by mmurto**
The create run form component has gotten quite large and hard to follow.
Simplify it a little by extracting the logic to functions, for which tests
can be written in a follow-up.

Signed-off-by: Mikko Murto <mikko.murto@doubleopen.org>

### CODE REVIEW

File: ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/-create-run-utils.ts [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+          ortRun.jobConfigContext || baseDefaults.jobConfigContext,
+      }
+    : baseDefaults;
+
+  return defaultValues;
+}
+
+/**
+ * Due to API schema and requirements for the form schema, the form values can't be directly passed
+ * to the API. This functions converts form values to correct payload to create an ORT run.
```
Comment (Etsija) [inline ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/-create-run-utils.ts:330]: Comment nit: "This functions" -> "This function"

File: ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/-create-run-utils.ts [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+  ortRun: OrtRun | null
+): z.infer<typeof createRunFormSchema> {
+  // Default values for the form: edit only these, not the defaultValues object.
+  const baseDefaults = {
+    revision: 'main',
+    path: '',
+    jobConfigs: {
+      analyzer: {
+        enabled: true,
+        repositoryConfigPath: '',
```
Comment (Etsija) [inline ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/-create-run-utils.ts:107]: Was the verdict that we should have `undefined` here?

Comment (mmurto) [reply ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/-create-run-utils.ts:107]: Probably yes. This PR only moves the existing logic to functions, so it's easier to work on and possibly write tests for.

**Review by Etsija — APPROVED** [review]
